core/domain/html_validation_service_test.py

```diff
@@ -1236,3 +1236,25 @@ def test_parsable_as_xml(self):
         self.assertEqual(
             html_validation_service.is_parsable_as_xml(valid_xml),
             True)
+
+    def test_convert_svg_diagram_tags_to_image_tags(self):
```

Comment:
I suggest also adding test cases for:
1. Multiple images to migrate (to ensure the loop is working)
2. Html with existing noninteractive-image tags (to ensure they aren't migrated)
3. Cases when the alt or image text is missing (I'm guessing these should result in failures?)

Reply:
Added test cases for the first two cases. Did not add for the third case as it won't arise, as the tag would always have the alt and image text.SVG diagrams can only be stored when both are provided currently.

Reply:
How can we be sure that (3) never happens? Was this already audited for production code? I'm checking because we often see special edge cases we don't anticipate, and it's sometimes a good idea to make sure those cases are explicitly handled even if they should never happen.

Reply:
We have a validation check in the front-end itself. Save button does not get enabled until and unless both of them are added.

Reply:
Ah, this additional context is really helpful. In that case, the current solution LGTM. Thanks!